SEUNAGBEYE/React-Redux-Es6
# #161368636 Error handling and notification

### What does this PR do?
Status and error handling

### Description of Task to be completed?
- add loading component
- show success toast when a request is successful
- show success toast when a request did not succeeds
### How should this be manually tested?
- fetch this branch: `git fetch originft-status-and-error-handling-161368636
- install all dependencies with `npm i`
- start the app `npm start -s`
- click on `Add Course` button to add a course
- fill in the course information and click on save
- then you will be redirected to the courses page where you will see the newly created course with a success notification
- to test for failure, repeat the step above but do not fill in the form value then click save

### Any background context you want to provide?
NA

### What are the relevant pivotal tracker stories?
[#161368636](https://www.pivotaltracker.com/n/projects/2200608/stories/161368636)

### Screenshots (if appropriate)
<img width="649" alt="screen shot 2018-10-24 at 1 44 02 am" src="https://user-images.githubusercontent.com/25517388/47398945-520f4500-d72e-11e8-83f0-c54a0d63724f.png">

<img width="643" alt="screen shot 2018-10-24 at 1 43 08 am" src="https://user-images.githubusercontent.com/25517388/47398927-31df8600-d72e-11e8-81b0-42554ea8671c.png">
